AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/Support/MutationEvent+Extensions.swift

```diff
@@ -30,21 +29,34 @@ extension MutationEvent {
                     return
                 }
 
-                if existingEvent.version == nil {
-                    Amplify.log.verbose("""
-                        Replacing existing mutation event having nil version with version from mutation response
-                            \(mutationSync.syncMetadata.version)
-                        """)
-                    existingEvent.version = mutationSync.syncMetadata.version
-                    storageAdapter.save(existingEvent, condition: nil) { result in
-                        switch result {
-                        case .failure(let dataStoreError):
-                            completion(.failure(dataStoreError))
-                        case .success:
-                            completion(.success(()))
+                // check if version returned from API is not nil and <= version sent in the request
```

Comment:
`// check if the pending mutation version is not nil and <= version from the response``

Reply:
```
guard let existingVersion = existingEvent.version, existingVersion <= mutationSync.syncMetadata.version else {
completion(.success(()))
   return
}
```